hash.c

```diff
@@ -1873,6 +1873,49 @@ rb_hash_s_try_convert(VALUE dummy, VALUE hash)
     return rb_check_hash_type(hash);
 }
 
+/*
+ *  call-seq:
+ *     Hash.ruby2_keywords?(hash) -> true or false
+ *
+ *  Check if a given hash is flagged by Module#ruby2_keywords (or
+ *  Proc#ruby2_keywords) or not.
+ *  This method is not for casual use; debugging, researching, and
+ *  some truly necessary cases like serialization of arguments.
+ *
+ *     ruby2_keywords def foo(*args)
+ *       Hash.ruby2_keywords?(args.last)
+ *     end
+ *     foo(k: 1)   #=> true
+ *     foo({k: 1}) #=> false
+ */
+static VALUE
+rb_hash_s_ruby2_keywords_p(VALUE dummy, VALUE hash)
+{
+    return (RHASH(hash)->basic.flags & RHASH_PASS_AS_KEYWORDS) ? Qtrue : Qfalse;
```

Comment:
Do you get `TypeError` when passing non-Hash instance as hash? I would guess no, but I didn't actually check.  I think `TypeError` should be raised in this case.  Same issue with `rb_hash_s_ruby2_keywords_bang` below.

Reply:
Good catch!  I've pushed the fix.  Thanks